modules/mobile/spec/helpers/mobile/appointments_cache_interface_spec.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Mobile::AppointmentsCacheInterface do
+  let(:subject) { described_class.new }
+  let(:user) { build(:user) }
+  let(:today) { '2020-11-01T10:30:00Z' }
+  # setting cache to empty array is sufficient for testing the "cache is set" behavior
+  let(:cached_data) { [] }
+  let(:mocked_appointments) { %i[appt1 appt2] }
+
+  def set_cache
+    Mobile::V0::Appointment.set_cached(user, cached_data)
+  end
+
+  describe '#fetch_appointments' do
+    describe 'cache fetching' do
+      context 'when fetch_cache is true' do
+        context 'and cache is set' do
+          before { set_cache }
+
+          it 'fetches data from the cache and does not request from upstream' do
+            expect(Mobile::V0::Appointment).to receive(:get_cached).and_call_original
+            expect(Mobile::V2::Appointments::Proxy).not_to receive(:new)
```

Comment:
I really wanted to avoid specifically testing against v2 for the caching tests by just ensuring that the AppointmentsCacheInterface#fetch_from_external_service method is called, which IMO is sufficient for these tests. But the linter shut me down.